pkg/webhook/sidecarset/validating/utils.go

```diff
@@ -46,5 +48,17 @@ func getCoreVolumes(volumes []v1.Volume, fldPath *field.Path) ([]core.Volume, fi
 func isSidecarSetNamespaceDiff(origin *appsv1alpha1.SidecarSet, other *appsv1alpha1.SidecarSet) bool {
 	originNamespace := origin.Spec.Namespace
 	otherNamespace := other.Spec.Namespace
-	return originNamespace != "" && otherNamespace != "" && originNamespace != otherNamespace
+	if originNamespace != "" && otherNamespace != "" && originNamespace != otherNamespace {
+		return true
+	}
+	originSelector := origin.Spec.NamespaceSelector
+	otherSelector := other.Spec.NamespaceSelector
+	if originSelector != nil && otherSelector != nil && !util.IsSelectorOverlapping(originSelector, otherSelector) {
+		return true
+	}
+	if (originNamespace != "" && otherSelector != nil) ||
+		(otherNamespace != "" && originSelector != nil) {
+		return true
```

Comment:
can we check other namesapce Selector can select the origin namespace before return true ?